openc3/lib/openc3/script/queue.rb

```diff
@@ -0,0 +1,100 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'openc3/script/extract'
+
+module OpenC3
+  module Script
+    include Extract
+
+    private
+
+    # Helper method that makes the request and parses the response
+    def _make_request(action:, verb:, uri:, scope:)
+      response = $api_server.request(verb, uri, scope: scope)
+      if response.nil?
+        raise "Failed to #{action} queue. No response from server."
+      elsif response.status != 200 and response.status != 201
+        result = JSON.parse(response.body, :allow_nan => true, :create_additions => true)
+        raise "Failed to #{action} queue due to #{result['message']}"
+      end
+      return JSON.parse(response.body, :allow_nan => true, :create_additions => true)
+    end
+
+    def queue_all(scope: $openc3_scope)
+      return _make_request(action: 'index', verb: 'get', uri: "/openc3-api/queues", scope: scope)
+    end
+
+    def queue_get(name, scope: $openc3_scope)
+      return _make_request(action: 'get', verb: 'get', uri: "/openc3-api/queues/#{name}", scope: scope)
+    end
+
+    def queue_list(name, scope: $openc3_scope)
+      return _make_request(action: 'list', verb: 'get', uri: "/openc3-api/queues/#{name}/list", scope: scope)
+    end
+
+    def queue_create(name, scope: $openc3_scope)
+      return _make_request(action: 'create', verb: 'post', uri: "/openc3-api/queues/#{name}", scope: scope)
+    end
+
+    def queue_hold(name, scope: $openc3_scope)
+      return _make_request(action: 'hold', verb: 'post', uri: "/openc3-api/queues/#{name}/hold", scope: scope)
+    end
+
+    def queue_release(name, scope: $openc3_scope)
+      return _make_request(action: 'release', verb: 'post', uri: "/openc3-api/queues/#{name}/release", scope: scope)
+    end
+
+    def queue_disable(name, scope: $openc3_scope)
+      return _make_request(action: 'disable', verb: 'post', uri: "/openc3-api/queues/#{name}/disable", scope: scope)
+    end
+
+    def queue_delete(name, scope: $openc3_scope)
+      return _make_request(action: 'delete', verb: 'delete', uri: "/openc3-api/queues/#{name}", scope: scope)
+    end
+    alias queue_destroy queue_delete
+
+    def queue_push(name, command, scope: $openc3_scope)
+      data = { command: command }
+      response = $api_server.request('post', "/openc3-api/queues/#{name}/push", data: data, json: true, scope: scope)
+      if response.nil?
+        raise "Failed to push command to queue. No response from server."
+      elsif response.status != 200
+        result = JSON.parse(response.body, :allow_nan => true, :create_additions => true)
+        raise "Failed to push command to queue due to #{result['message']}"
+      end
+      return JSON.parse(response.body, :allow_nan => true, :create_additions => true)
+    end
```

Comment:
Not sure I really need `queue_push` because normally you're interacting with it via the `queue` option in `cmd`.